crates/core/types/transaction.rs

```diff
@@ -132,6 +132,14 @@ impl RLPDecode for Transaction {
     /// A) `TransactionType || Transaction` (Where Transaction type is an 8-bit number between 0 and 0x7f, and Transaction is an rlp encoded transaction of type TransactionType)
     /// B) `LegacyTransaction` (An rlp encoded LegacyTransaction)
     fn decode_unfinished(rlp: &[u8]) -> Result<(Self, &[u8]), RLPDecodeError> {
+        // Adjust the encoding in case the transaction is encoded as a string (prefix between 0xb8 and 0xbf)
```

Comment:
when does this happen?

Reply:
In the ef_tests all the non-legacy transactions are encoded like this. You can see it for example in the test `eip5656_mcopy/mcopy_memory_expansion/mcopy_huge_memory_expansion.json`.

Reply:
Ok, I think this is the standard way of encoding EIP-2718 transactions, the previous implementation was buggy.

So the logic should be:
- If the transaction is encoded as bytes (prefix between 0xb8 and 0xbf) it means it is a EIP-2718
- Otherwise it is a legacy transaction

A transaction that is not legacy and is not encoded as bytes (prefix between 0xb8 and 0xbf) should fail. <-- this is not validated right now.

Reply:
Done!